test/e2e/apps/deployment.go

```diff
@@ -134,6 +141,238 @@ var _ = SIGDescribe("Deployment", func() {
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			deployment, ok := event.Object.(*appsv1.Deployment)
+			framework.ExpectEqual(ok, true, "unable to convert event.Object type")
+			if deployment.Status.AvailableReplicas == testDeploymentDefaultReplicas &&
+				deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("patching the Deployment")
+		deploymentPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{"test-deployment": "patched"},
+			},
+			"spec": map[string]interface{}{
+				"replicas": testDeploymentMinimumReplicas,
+				"template": map[string]interface{}{
+					"spec": map[string]interface{}{
+						"containers": [1]map[string]interface{}{{
+							"name":    testDeploymentName,
+							"image":   testDeploymentPatchImage,
+							"command": []string{"/bin/sleep", "100000"},
+						}},
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to Marshal Deployment JSON patch")
+		_, err = f.ClientSet.AppsV1().Deployments(testNamespaceName).Patch(context.TODO(), testDeploymentName, types.StrategicMergePatchType, []byte(deploymentPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Deployment")
+
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Modified {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("waiting for Replicas to scale")
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			deployment, ok := event.Object.(*appsv1.Deployment)
+			framework.ExpectEqual(ok, true, "unable to convert event.Object type")
+			if deployment.Status.AvailableReplicas == testDeploymentMinimumReplicas &&
+				deployment.Status.ReadyReplicas == testDeploymentMinimumReplicas {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("listing Deployments")
+		deploymentsList, err := f.ClientSet.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{LabelSelector: testDeploymentLabelsFlat})
+		framework.ExpectNoError(err, "failed to list Deployments")
+		foundDeployment := false
+		for _, deploymentItem := range deploymentsList.Items {
+			if deploymentItem.ObjectMeta.Name == testDeploymentName &&
+				deploymentItem.ObjectMeta.Namespace == testNamespaceName &&
+				deploymentItem.ObjectMeta.Labels["test-deployment-static"] == "true" &&
+				*deploymentItem.Spec.Replicas == testDeploymentMinimumReplicas &&
+				deploymentItem.Spec.Template.Spec.Containers[0].Image == testDeploymentPatchImage {
+				foundDeployment = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundDeployment, true, "unable to find the Deployment in list", deploymentsList)
```

Comment:
I might swap this with the loop above.  Verify that patch has taken effect on the deployment spec before waiting for its status to reconcile.

Reply:
Listing the Deployments in all namespaces here was originally put in here so the endpoint would be hit but also running checks since we have some data. I think it's OK to swap it out as you suggest, being as the same call is made at the start of the test (to get the beginning resource version)

Reply:
However, it might be good to still have the functionality of still finding the deployment, created in this test, in a list of all deployments in all namespaces.